Sources/ProjectDescription/PackageSettings.swift

```diff
@@ -26,6 +26,21 @@ import Foundation
 /// )
 /// ```
 public struct PackageSettings: Codable, Equatable {
+    /**
+     SPM tries to make linking work by making build-time decisions like how to link packages on behalf of users.
+     While this is desirable at a small scale (it's convenient), we believe it's not a good idea at a large scale because it might
+     cause issues like duplicated symbols or increased binary size.
+
+     We believe that how things are linked is something that Xcode project maintainers should think about, with Tuist helping
+     with any complexities associated with it, but we acknowledge the interest of some of our users to align with SPM's convenient
+     behaviour, and therefore we include this flag for users to opt-into that behaviour.
+
+     Here are some things that we do as part of that.
+
+     - We set the `GENERATE_MASTER_OBJECT_FILE` build setting in static targets by default (https://github.com/apple/swift/issues/48561)
+     */
+    public var spmLinkingStyle: Bool
```

Comment:
Is this meant to continually map to the linking approach of SPM? Is there anything else that could describe what this actually controls?

Mostly asking around naming if this is going to control more than just `generate_master_object_file` only long term or more settings if they're needed.  Perhaps `prelinkSingleObjectFile` or `generateMasterObjectFile` to be more explicit on the one setting it controls and potentially allow for more granular flexibility in the future.

Reply:
@fortmarek & @danieleformichelli would like to mimic the mapping that SPM does from "automatic" linking to "static/dynamic", hence the idea of picking a more generic name, but we can also take a more granular naming approach and let users opt-in/out from every configuration. My concern is that it can easily become a flat and broad surface of criptic configuration flags.

Reply:
I'd rather not add this at the current stage. If SPM sets `GENERATE_MASTER_OBJECT_FILE`, I'd set it, too. Our default setting should imho make the best effort of ensuring libraries added by users compile and work, while ensuring developers have a way to tinker with low-level settings if they need to optimise their binary size, etc.

In this case, we'd still allow anybody to change this setting in their `PackageSettings` via `baseSettings`. What we can also consider is adding a `generationOption` property of `PackageSettings` that would make it easier to disable the extra convenience configuration that we'd be adding:

```swift
let packageSettings = PackageSettings(
  generationOptions: [
   .disableGenerateMasterObjectFile,
   // if we decide to respect .automatic linking type, we could also do
   .enableAutomaticLinking // Or `.disableAutomaticLinking` if that would ever become the default
  ]
)
```